ocdskingfisher/collection_util.py

```diff
@@ -0,0 +1,59 @@
+import ocdskingfisher.database
+import ocdskingfisher.sources_util
+import os
+
+
+class Collection:
+
+    def __init__(self, source_id=None, data_version=None, sample=None, on_disk=None, database_id=None):
+        self.source_id = source_id
+        self.data_version = data_version
+        self.sample = sample
+        self.database_id = database_id
+        self.on_disk = on_disk
+
+
+def get_all_collections(data_dir):
+    out = []
+
+    # Load from database
+    for result in ocdskingfisher.database.get_all_collections():
+        out.append(Collection(
+            database_id=result['id'],
+            source_id=result['source_id'],
+            data_version=result['data_version'],
+            sample=result['sample'],
+        ))
+
+    # Load from disk
+    for sample in [True, False]:
+        for key in ocdskingfisher.sources_util.gather_sources().keys():
+            directory = os.path.join(data_dir, key + ("_sample" if sample else ""))
+            if os.path.isdir(directory):
+                for version in os.listdir(directory):
+                    if os.path.isdir(os.path.join(directory, version)) and \
+                            os.path.isfile(os.path.join(directory, version, "metadb.sqlite3")):
+                        _get_all_collections_disk_collection_found(
+                            source_id=key,
+                            data_version=version,
+                            sample=sample,
+                            out=out,
+                        )
+
+    # Return results
```

Comment:
Would be nice if the results were sorted by source_id so that all collections from same source are together.

Reply:
My guess was that there would be different orderings people would want, so maybe we'd add a command line flag later. Tho maybe if we checked out the user cases only ordering by source would come up - I didn't give it too much thought, I admit.